addons/iap/models/iap_account.py

```diff
@@ -17,103 +18,122 @@
 
-    def _search_info(self, operator, value):
-        return []
+    def web_save(self, *args, **kwargs):
+        return super(IapAccount, self.with_context(disable_iap_fetch=True)).web_save(*args, **kwargs)
 
     def write(self, values):
-        res = super(IapAccount, self).write(values)
-        iap_edits = ['warn_me', 'warning_threshold', 'warning_email']
-        if any(edited_attribute in values for edited_attribute in iap_edits):
-            try:
-                route = '/iap/update-warning-odoo'
-                endpoint = iap_tools.iap_get_endpoint(self.env)
-                url = endpoint + route
+        res = super().write(values)
+        if (
+            not self.env.context.get('disable_iap_update')
+            and any(warning_attribute in values for warning_attribute in ('warning_threshold', 'warning_user_ids'))
+        ):
+            route = '/iap/1/update-warning-email-alerts'
+            endpoint = iap_tools.iap_get_endpoint(self.env)
+            url = werkzeug.urls.url_join(endpoint, route)
+            for account in self:
                 data = {
-                    'account_token': self.mapped('account_token')[0],
-                    'dbuuid': self.env['ir.config_parameter'].sudo().get_param('database.uuid'),
-                    'warn_me': values.get('warn_me'),
-                    'warning_threshold': values.get('warning_threshold'),
-                    'warning_email': values.get('warning_email'),
+                    'account_token': account.account_token,
+                    'warning_threshold': account.warning_threshold,
+                    'warning_emails': [{
+                        'email': user.email,
+                        'lang_code': user.lang or get_lang(self.env).code,
+                    } for user in account.warning_user_ids],
                 }
-                iap_tools.iap_jsonrpc(url=url, params=data)
-            except AccessError as e:
-                _logger.warning('Save service error : %s', str(e))
+                try:
+                    iap_tools.iap_jsonrpc(url=url, params=data)
+                except AccessError as e:
+                    _logger.warning("Update of the warning email configuration has failed: %s", str(e))
         return res
 
-    def get_services(self):
+    @staticmethod
+    def is_running_test_suite():
+        return hasattr(threading.current_thread(), 'testing') and threading.current_thread().testing
+
+    def _get_account_information_from_iap(self):
+        # During testing, we don't want to call the iap server
+        if self.is_running_test_suite():
+            return
+        route = '/iap/1/get-accounts-information'
+        endpoint = iap_tools.iap_get_endpoint(self.env)
+        url = werkzeug.urls.url_join(endpoint, route)
+        params = {
+            'iap_accounts': [{
+                'token': account.account_token,
+                'service': account.service_id.technical_name,
+            } for account in self],
```

Comment:
We should check that the account has a `service_id` set.
Even though it is a required field, at creation time, this function is called while the `service_id` is unset.
It causes a `ValueError: False service not found` on IAP side.